opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -2295,6 +2295,74 @@ namespace detail {
 
 
 
+
+    template <class Grid, class WellModel, class Implementation>
+    std::vector<V>
+    BlackoilModelBase<Grid, WellModel, Implementation>::
+    computeFluidInPlace(const ReservoirState& x,
+                        const WellState& xw,
+                        const std::vector<int>& fipnum)
+    {
+        using namespace Opm::AutoDiffGrid;
+        const int nc = numCells(grid_);
+        const int np = x.numPhases();
+
+        SolutionState state(np);
+        const DataBlock s = Eigen::Map<const DataBlock>(& x.saturation()[0], nc, np);
+        state.pressure    = ADB::constant(Eigen::Map<const V>(& x.pressure()[0], nc, 1));
+        state.temperature = ADB::constant(Eigen::Map<const V>(& x.temperature()[0], nc, 1));
+        state.saturation[Water] = ADB::constant(s.col(Water));
+        state.saturation[Oil] = ADB::constant(s.col(Oil));
+        state.saturation[Gas] = ADB::constant(s.col(Gas));
+        state.rs =  ADB::constant(Eigen::Map<const V>(& x.gasoilratio()[0], nc, 1));
+        state.rv = ADB::constant(Eigen::Map<const V>(& x.rv()[0], nc, 1));
+        state.canonical_phase_pressures = computePressures(state.pressure, 
+                                                           state.saturation[Water], 
+                                                           state.saturation[Oil], 
+                                                           state.saturation[Gas]);
+
+        const Opm::PhaseUsage& pu = fluid_.phaseUsage();
+
+        const std::vector<PhasePresence> cond = phaseCondition();
+
+
+
+        const ADB pv_mult = poroMult(state.pressure);
+        const V& pv = geo_.poreVolume();
+        const int maxnp = Opm::BlackoilPhases::MaxNumPhases;
+        std::vector<V> fip(5, V::Zero(nc));
+        for (int phase = 0; phase < maxnp; ++phase) {
+            if (active_[ phase ]) {
+                const int pos = pu.phase_pos[ phase ];
+                const auto& b = asImpl().fluidReciprocFVF(phase, state.canonical_phase_pressures[phase], state.temperature, state.rs, state.rv, cond);
```

Comment:
Even if we cannot directly use the accum[] data, we can use the stored rq_[phase].b data here, right?